clustered/client/src/main/java/org/ehcache/clustered/client/internal/ConnectionSource.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.client.internal;
+
+import org.terracotta.connection.ConnectionException;
+import org.terracotta.lease.connection.LeasedConnection;
+import org.terracotta.lease.connection.LeasedConnectionFactory;
+
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+
+public abstract class ConnectionSource {
+
+  public abstract String getCacheManagerName();
+
+  public abstract LeasedConnection connect(Properties connectionProperties) throws ConnectionException;
+
+  public abstract URI getClusterUri();
+
+  public abstract Iterable<InetSocketAddress> getServers();
+
+  public abstract String toString();
+
+  public static class ClusterUri extends ConnectionSource {
+
+    private final URI clusterUri;
+    private final String cacheManagerName;
+
+    public ClusterUri(URI clusterUri) {
+      this.clusterUri = Objects.requireNonNull(clusterUri, "Cluster URI cannot be null");
+      this.cacheManagerName = extractCacheManager(clusterUri);
```

Comment:
I think it is wrong to use `cacheManagerName` because a cache manager does not have a name. Except for management, where we have an alias that enables us to identify the client part but this is a specific use case. The cache manager is only linked through the URI to an `entityName`. And the entity name could be something completely different than the cache manager alias set in management.

When connecting to a cluster, the cluster clientId used is composed of the entityName:

`connectionProperties.put(ConnectionPropertyNames.CONNECTION_NAME, CONNECTION_PREFIX + entityIdentifier);`

So I think yes it should be renamed everywhere because I think using a variable name `cacheManagerName` is wrong semantically speaking. 

But I'll let @chrisdennis and @cljohnso  and @cschanck decide ;-)

Reply:
@mathieucarbou I concur.  The two terms we tend to use internally (or we should use) are "clusterTierManager" when describing the thing that you are addressing here, and "entityIdentifier" the referring to the Voltron specific concept. I would lean towards referring to the `clusterTierManager` in public methods, and the `entityIdentifier` when moving closer to the implementation and the actual Voltron code.